common/post-fs-data.sh

```diff
@@ -1,18 +1,76 @@
+    local LN=$(sed -n "/<$MODID\/>/=" $2)
+    for i in ${LN}; do
+      sed -i "$i d" $2
+      case $(sed -n "$((i-1)) p" $2) in
+        *">$MODID-->") sed -i -e "${i-1}s/<!--$MODID-->//" -e "${i-1}s/$/<!--$MODID-->/" $2;;
+        *) sed -i "$i p" $2
+           sed -ri "${i}s/(^ *)(.*)/\1<!--$MODID\2$MODID-->/" $2
+           sed -i "$((i+1))s/$/<!--$MODID-->/" $2;;
+      esac
+    done
+    case "$1" in
+      "-u"|"-s") xmlstarlet ed -L -u "$3/@$VALC" -v "$VAL" $2;;
+      "-d") xmlstarlet ed -L -d "$3" $2;;
+    esac
+  elif [ "$1" == "-s" ]; then
+    local NP=$(echo "$3" | sed -r "s|(^.*)/.*$|\1|")
+    local SNP=$(echo "$3" | sed -r "s|(^.*)\[.*$|\1|")
+    local SN=$(echo "$3" | sed -r "s|^.*/.*/(.*)\[.*$|\1|")
+    xmlstarlet ed -L -s "$NP" -t elem -n "$SN-$MODID" -i "$SNP-$MODID" -t attr -n "$NAMEC" -v "$NAME" -i "$SNP-$MODID" -t attr -n "$VALC" -v "$VAL" $2
+    xmlstarlet ed -L -r "$SNP-$MODID" -v "$SN" $2
+    xmlstarlet ed -L -i "$3" -t elem -n "$MODID" $2
+    local LN=$(sed -n "/<$MODID\/>/=" $2)
+    for i in ${LN}; do
+      sed -i "$i d" $2
+      sed -ri "${i}s/$/<!--$MODID-->/" $2
+    done 
+  fi
+  local LN=$(sed -n "/^ *<!--$MODID-->$/=" $2 | tac)
+  for i in ${LN}; do
+    sed -i "$i d" $2
+    sed -ri "$((i-1))s/$/<!--$MODID-->/" $2
+  done 
+}
+
+(
+if [ $API -ge "28" ]; then
+  DPF=/data/data/com.google.android.dialer/shared_prefs/dialer_phenotype_flags.xml
+  if [ -f $DPF ]; then
     # Enabling Google's Call Screening
-    sed -i 's/name="G__speak_easy_bypass_locale_check" value="false"/name="G__speak_easy_bypass_locale_check" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__speak_easy_enable_listen_in_button" value="false"/name="G__speak_easy_enable_listen_in_button" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="__data_rollout__SpeakEasy.OverrideUSLocaleCheckRollout__launched__" value="false"/name="__data_rollout__SpeakEasy.OverrideUSLocaleCheckRollout__launched__" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__enable_speakeasy_details" value="false"/name="G__enable_speakeasy_details" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__speak_easy_enabled" value="false"/name="G__speak_easy_enabled" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__speakeasy_show_privacy_tour" value="false"/name="G__speakeasy_show_privacy_tour" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="__data_rollout__SpeakEasy.SpeakEasyDetailsRollout__launched__" value="false"/name="__data_rollout__SpeakEasy.SpeakEasyDetailsRollout__launched__" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="__data_rollout__SpeakEasy.CallScreenOnPixelTwoRollout__launched__" value="false"/name="__data_rollout__SpeakEasy.CallScreenOnPixelTwoRollout__launched__" value="true"/g' $DIALER_PREF_FILE
-    sed -i 's/name="G__speakeasy_postcall_survey_enabled" value="false"/name="G__speakeasy_postcall_survey_enabled" value="true"/g' $DIALER_PREF_FILE
+    #DIALERPATCHES
```

Comment:
I assume this is being worked on.

Reply:
Nope. This is fine. The install.sh copy the patch_xml function copys the patchs to #DIALERPATCHES in post-fs-data

Reply:
case $2 in
    *dialer_phenotype_flags*.xml)  sed -i "/#DIALERPATCHES/a\          patch_xml $1 \$MODPATH/\ '$3' \"$4\"" $INSTALLER/common/post-fs-data.sh; VAR1=boolean; VAR2=string; VAR3=long;;

Reply:
Ah cool